pallets/liquid-staking/src/lib.rs

```diff
@@ -785,194 +751,70 @@ pub mod pallet {
             value: BalanceOf<T>,
             payee: RewardDestination<T::AccountId>,
         ) -> DispatchResult {
-            let stash = Self::derivative_para_account_id();
-            let controller = stash.clone();
-
-            switch_relay!({
-                let call =
-                    RelaychainCall::Utility(Box::new(UtilityCall::BatchAll(UtilityBatchAllCall {
-                        calls: vec![
-                            RelaychainCall::Balances(BalancesCall::TransferKeepAlive(
-                                BalancesTransferKeepAliveCall {
-                                    dest: T::Lookup::unlookup(stash),
-                                    value,
-                                },
-                            )),
-                            RelaychainCall::Utility(Box::new(UtilityCall::AsDerivative(
-                                UtilityAsDerivativeCall {
-                                    index: T::DerivativeIndex::get(),
-                                    call: RelaychainCall::Staking::<T>(StakingCall::Bond(
-                                        StakingBondCall {
-                                            controller: T::Lookup::unlookup(controller.clone()),
-                                            value,
-                                            payee: payee.clone(),
-                                        },
-                                    )),
-                                },
-                            ))),
-                        ],
-                    })));
-
-                let msg = Self::ump_transact(call.encode().into(), Self::xcm_weight().bond_weight)?;
-
-                match T::XcmSender::send_xcm(MultiLocation::parent(), msg) {
-                    Ok(()) => {
-                        Self::deposit_event(Event::<T>::Bonding(controller, value, payee));
-                    }
-                    Err(_e) => {
-                        return Err(Error::<T>::BondFailed.into());
-                    }
-                }
-            });
-
-            Ok(())
+            T::XCM::bond_internal(
```

Comment:
bond_internal can be renamed to bond now, and keep the internal mutation else where in pallet-liquid-staking

Reply:
@GopherJ Can we do this as a part of separate issue?